examples/pytorch/infograph/unsupervised.py

```diff
@@ -0,0 +1,144 @@
+        args.device = 'cuda:{}'.format(args.gpu)
+    else:
+        args.device = 'cpu'
+
+    return args
+
+    
+def collate(samples):
+    ''' collate function for building graph dataloader'''
+    
+    graphs, labels = map(list, zip(*samples))
+
+    # generate batched graphs and labels
+    batched_graph = dgl.batch(graphs)
+    batched_labels = th.tensor(labels)
+
+    n_graphs = len(graphs)
+    graph_id = th.arange(n_graphs)
+    graph_id = dgl.broadcast_nodes(batched_graph, graph_id)
+
+    batched_graph.ndata['graph_id'] = graph_id
+
+    return batched_graph, batched_labels
+
+
+if __name__ == '__main__':
+
+    # Step 1: Prepare graph data   ===================================== #
+    args = argument()
+    print(args)
+
+    log_interval = 1
+
+    # load dataset from dgl.data.GINDataset
+    dataset = GINDataset(args.dataname, self_loop = False)
+
+    # get graphs and labels
+    graphs, labels = map(list, zip(*dataset))
+
+    # generate a full-graph with all examples for evaluation
+    wholegraph = dgl.batch(graphs)
+    wholegraph.ndata['attr'] = wholegraph.ndata['attr'].to(th.float32)
+
+    # create dataloader for batch training
+    dataloader = GraphDataLoader(dataset,
+                                 batch_size=args.batch_size,
+                                 collate_fn=collate,
+                                 drop_last=False,
+                                 shuffle=True)
+
+    in_dim = wholegraph.ndata['attr'].shape[1]
+
+    # Step 2: Create model =================================================================== #
+    model = InfoGraph(in_dim, args.hid_dim, args.n_layers)
+    model = model.to(args.device)
+
+    # Step 3: Create training components ===================================================== #
+    optimizer = th.optim.Adam(model.parameters(), lr=args.lr)
+ 
+    print('===== Before training ======')
+    
+    wholegraph = wholegraph.to(args.device)
+    wholefeat = wholegraph.ndata['attr']
+    
+    emb = model.get_embedding(wholegraph, wholefeat).cpu()
+    res = evaluate_embedding(emb, labels, args.device)
+
+    print('logreg {:4f}, svc {:4f}'.format(res[0], res[1]))
+    
+    best_logreg = 0
+    best_svc = 0
+    best_epoch = 0
+    best_loss = 0
+
+    # Step 4: training epochs =============================================================== #
+    for epoch in range(1, args.epochs):
+        loss_all = 0
+        model.train()
+    
+        for graph, label in dataloader:
+            
+            graph = graph.to(args.device)
+            feat = graph.ndata['attr']
+            graph_id = graph.ndata['graph_id']
+            
+            n_graph = label.shape[0]
+    
+            optimizer.zero_grad()
+            loss = model(graph, feat, graph_id)
+            loss.backward()
+            optimizer.step()
+            loss_all += loss.item() * n_graph
+    
+        print('Epoch {}, Loss {:.4f}'.format(epoch, loss_all / len(dataloader)))
```

Comment:
Do you want to divide `loss_all` by the number of batches or the number of graphs?